src/hb-ot-shape-fallback.cc

```diff
@@ -525,6 +525,13 @@ _hb_ot_shape_fallback_spaces (const hb_ot_shape_plan_t *plan HB_UNUSED,
   for (unsigned int i = 0; i < count; i++)
     if (_hb_glyph_info_is_unicode_space (&info[i]) && !_hb_glyph_info_ligated (&info[i]))
     {
+      /* If font had no ASCII space and we used the invisible glyph, give it a 1/4 EM default advance. */
+      if (buffer->invisible && info[i].codepoint == buffer->invisible)
+        if (horizontal)
+	  pos[i].x_advance = +font->x_scale / 4;
+        else
+	  pos[i].y_advance = -font->y_scale / 4;
+
```

Comment:
If I read this correctly, you are giving  0x20 a width of em/4 here, while https://docs.microsoft.com/en-us/typography/develop/character-design-standards/whitespace says it should be em/2.

Indeed, in testing, the result of this code is half as wide as what my current pango code produces.

Reply:
IMO em/2 sounds unusually large compared to what most fonts use.

Indeed, the MS page mentioned actually says

> For an *average* width font a good value is ~1/4 the em.

Reply:
ah, ok. Lets go with that then.